src/components/RadioButtonGroup/index.js

```diff
@@ -45,8 +61,62 @@ class RadioButtonGroup extends Component {
         return undefined;
     }
 
+    getCheckedOptionRef() {
+        const { value, options } = this.props;
+        const currentOptionIndex = options.findIndex(option => isOptionChecked(option, value));
+        return currentOptionIndex !== -1 ? this.optionsRefs[currentOptionIndex] : null;
+    }
+
+    generateRefsForOptions() {
+        const { options } = this.props;
+        return options.reduce(list => [...list, React.createRef()], []);
+    }
+
+    addRefsToOptions(options) {
+        return options.reduce(
+            (list, option, index) => [
+                ...list,
+                {
+                    optionRef: this.optionsRefs[index],
+                    ...option,
+                },
+            ],
+            [],
+        );
+    }
+
+    isMarkerActive() {
+        const { value, options } = this.props;
+        const reducer = (accumulator, option) => accumulator || isOptionChecked(option, value);
+
+        return options.reduce(reducer, false);
+    }
+
+    handleChange(event) {
+        const { onChange } = this.props;
+        onChange(event);
+        setTimeout(() => {
+            this.updateMarker();
+        }, 0);
```

Comment:
handleChange is not needed, pass onChange direct to ButtonItems,
instead of update the marker here let's update it in componentDidUpdate when value change:
```
   componentDidUpdate(prevProps) {
        if (prevProps.value !== this.props.value) {
            this.updateMarker();
        }
    }
```